runtime/src/bank_forks.rs

```diff
@@ -246,7 +246,14 @@ impl BankForks {
             let context = SchedulingContext::new_with_mode(mode, bank.clone());
             let scheduler = scheduler_pool.take_scheduler(context);
             let bank_with_scheduler = BankWithScheduler::new(bank, Some(scheduler));
-            scheduler_pool.register_timeout_listener(bank_with_scheduler.create_timeout_listener());
+            if matches!(mode, SchedulingMode::BlockVerification) {
+                // Skip registering for block production. Both the replay stage and PohRecorder
+                // don't support _concurrent block production_ at all. It's strongly assumed that
+                // block is produced in singleton way and it's actually desired, while ignoring the
+                // opportunity cost of (hopefully rare!) fork switching...
```

Comment:
@steviez do we have data on how often we switch forks during leader slot? I'm not aware of any, but I don't believe it's **super** uncommon for it to happen?

Reply:
I don't think we have any data on hand, any metrics to easily measure this unfortunately